src/doc/layer.h

```diff
@@ -128,6 +129,9 @@ class Layer : public WithUserData {
   int opacity() const { return m_opacity; }
   void setOpacity(int opacity) { m_opacity = opacity; }
 
+  const base::Uuid& uuid() const { return m_uuid; }
+  void setUuid(const base::Uuid& uuid) { m_uuid = uuid; }
```

Comment:
One little nice check that we can add on debug mode is that we can set the UUID of the layer just one time if the UUID is generated inside `uuid()`:
```suggestion
  void setUuid(const base::Uuid& uuid)
  {
    ASSERT(m_uuid == base::Uuid());
    m_uuid = uuid;
  }
```